backend/internal/bumbal/handler_helpers.go

```
@@ -41,10 +43,115 @@ func requestBumbalActivity(w http.ResponseWriter, r *http.Request, reqBody []byt
+	// Make a request to Bumbal to retrieve the number of activities
+	reqBody := []byte(`{"count_only":true}`)
+	resp, err := requestBumbalActivity(w, r, reqBody)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return nil, err
+	}
+
+	// Check the status codes of the response from Bumbal
+	switch resp.StatusCode {
+	case http.StatusOK:
+		// If the response status is OK, get the data from the response's body
+		respModel, err := getResponseData(resp)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return nil, err
+		}
+
+		// Get the total number of needed requests
+		var numberOfRequests int32
+		if respModel.CountFiltered%100 == 0 {
+			numberOfRequests = respModel.CountFiltered / 100
+		} else {
+			numberOfRequests = respModel.CountFiltered/100 + 1
+		}
+
+		// Construct the layout of the requests to get activities
+		reqBodyHead := []byte(`{"options":{"include_address_applied":true,"include_depot_address": true},
+						"sorting_column":"id","as_list": true,"count_only":false,"limit":100,"offset":`)
+		reqBodyTail := []byte(`}`)
+
+		// Make a variable that will store a list of all activities collected from all requests
+		var fullRespModel models.ActivityListResponseBumbal
+		emptyList := make([]models.ActivityModelBumbal, 0)
+		fullRespModel.Items = &emptyList
+
+		// Set up a channel for collecting the responses from each request
+		respChan := make(chan *http.Response, numberOfRequests)
+		var mutex sync.Mutex  // mutex for protecting the channel
+		var wg sync.WaitGroup // wait group that waits for the goroutines to finish
+
+		for i := 0; i < int(numberOfRequests); i++ {
+			wg.Add(1)
+			// go routine that will make one of the requests to Bumbal
+			go func(i int) {
+				defer wg.Done()
+				offset := []byte(fmt.Sprintf("%d", i*100))
+				reqBody = append(append(reqBodyHead, offset...), reqBodyTail...)
+
+				partResponse, _ := requestBumbalActivity(w, r, reqBody)
```

Don't ignore error...

---

Here I am just collecting the responses, I take care of them later in the code anyway